### PR TITLE
Adds ambient occlusion

### PR DESCRIPTION
**Pull request by MoXuT0**
## About The Pull Request

Per title. Removes river overlays because they fuck with filters so much.

## Changelog
:cl:
add: Ambient occlusion.
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -918,6 +919,12 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			else if(!usr.client.tooltips && tooltips)
 				usr.client.tooltips = new /datum/tooltip(usr.client)
 
+		if("ambientocclusion")
+			ambientocclusion = !ambientocclusion
+			if(parent && parent.screen && parent.screen.len)
```
Comment (DominikPanic) [inline code/modules/client/preferences.dm]:
```suggestion
			if(length(parent?.screen))
```

File: code/_onclick/hud/plane_master.dm [hunk 2]
```diff
@@ -60,6 +60,8 @@
 
 /obj/screen/plane_master/game_world/backdrop(mob/mymob)
 	filters = list()
+	if(istype(mymob) && mymob.client && mymob.client.prefs && mymob.client.prefs.ambientocclusion)
```
Comment (DominikPanic) [inline code/_onclick/hud/plane_master.dm]:
```suggestion
	if(istype(mymob) && mymob.client?.prefs?.ambientocclusion)
```

**Comment by DominikPanic** [comment]
Did you manage to fix the issues with catwalks and corners?

**Comment by MoXuT0** [comment]
Still working on that

**Comment by MoXuT0** [comment]
I've fixed corner underlays but there's now another problem with overlays for river. Same shit as catwalks, dunno how to fix it.

river

**Comment by Rohesie** [comment]
Pref for ambient occlusion or not? Interesting.

**Comment by MoXuT0** [comment]
Only thing I thought of is too make river overlay apply on mob when you enter river and making pipes, cables and etc to be on `FLOOR_PLANE`. Don’t know how good this would be

**Comment by Rohesie** [comment]
Wonder how expensive that river-on-mob overlay it'd be.
We can test, I suppose.

**Comment by MoXuT0** [comment]
I've removed those ugly river overlays and brough back planes for pipes and weeds because otherwise it will look retarded.

**Comment by DominikPanic** [comment]
Bringing planes for weeds brought back several issues. Did you test the cases that were problematic before? (weeds going under walls, etc)

**Comment by MoXuT0** [comment]
I'll see if they are still present

**Comment by MoXuT0** [comment]
Welp, fuck it. They've appeared underwalls. And if just remove weeds plane, vents would appear under them. God I hate planes.

**Pull request closed without merging**